woocommerce/woocommerce-ios
# Orders: Display split view on large screens

Part of #6381 


### Description

This PR updates the Orders tab to show split view on large screens to take advantage of the available real estate.

The solution includes:
- Add a custom split view controller to be reused within the app. This split view controller uses the modern API to configure a double-column layout with the primary column being displayed when collapsed.
- Update the root view of the Orders tab to a new view controller that wraps the old order root within a split view controller. This is needed because the modern split view controller (initiated with `style`) doesn't work properly when added directly to a tab view controller.
- Extract `GhostTableViewController` to be a reusable type that displays ghost animation over a table view.
- Set up the orders split view controller with the order root and a ghost table view controller.
- Update the order list to automatically update the split view's detail view after loading orders
- Replace the order detail storyboard with a nib file and update the order detail view controller's initializer for easy setup.
- Additionally: update constraints on order root to make the view expand edge-to-edge in landscape mode.
- Also: updated UI tests to reflect the change to the Orders tab

### Testing instructions

On iPad (or large iPhones in landscape mode):
- Log in to a valid account. Notice that the details for the Orders tab in the tab bar are correct.
- Select the Orders tab. Notice that the list is displayed as a split view, with a placeholder view for the 2nd column initially.
- After a while, notice that details for the first item in the list are displayed on the 2nd column.
- Select another item, notice that the correct details are displayed on the 2nd column.

On iPhone (with small size):
- Log in to a valid account. Notice that the details for the Orders tab in the tab bar are correct.
- Select the Orders tab. Notice that the order list is displayed correctly.
- Select any order, notice that the correct details are displayed.

### Screenshots

On iPad:
https://user-images.githubusercontent.com/5533851/157225577-8f6aa477-867a-452c-90d6-665449408108.mp4

On iPhone:
https://user-images.githubusercontent.com/5533851/157227051-67d31080-cb80-4693-b886-c491c27bba19.mp4



---
